storage/zynq7000-flash/zynq7000-flash.c

```diff
@@ -0,0 +1,529 @@
+			break;
+
+		case mtMount:
+			mnt = (mount_msg_t *)msg->i.raw;
+			msg->o.attr.err = storage_mountfs(storage_get(GET_STORAGE_ID(mnt->id)), mnt->fstype, NULL, 0, (oid_t *)msg->o.raw);
+			break;
+
+		case mtDevCtl:
+			msg->o.io.err = flash_devCtl(&msg->i.io.oid);
+			break;
+
+		default:
+			msg->o.io.err = -EINVAL;
+			break;
+	}
+}
+
+
+static void flash_help(const char *prog)
+{
+	printf("Usage: %s [options] or no args to automatically detect and initialize a NOR flash device\n", prog);
+	printf("\t-p <dev:start:size>          - register partition\n");
+	printf("\t\tdev:    device path\n");
+	printf("\t\tstart:  partition start in bytes\n");
+	printf("\t\tsize:   partition size in bytes\n");
+	printf("\t-r <dev:start:size:fs>       - mount root filesystem\n");
+	printf("\t\tdev:    device name\n");
+	printf("\t\tstart:  partition start in bytes\n");
+	printf("\t\tsize:   partition size in bytes\n");
+	printf("\t\tfs:     filesystem name\n");
+	printf("\t-h                           - print this help message\n");
+}
+
+
+static int flash_oidResolve(const char *devPath, oid_t *oid)
+{
+	int res;
+	char temp[32];
+
+	if (strncmp("/dev/", devPath, 5) != 0) {
+		return -EINVAL;
+	}
+
+	memset(temp, 0, sizeof(temp));
+	strcpy(temp, "devfs/");
+	strcpy(temp + 6, devPath + 5);
+
+	res = lookup(temp, NULL, oid);
+	if (res < 0) {
+		return res;
+	}
+
+	return res;
+}
+
+
+static int flash_createMtdDev(const storage_t *strg, oid_t *oid)
+{
+	int res;
+	char path[32];
+	storage_t *part;
+	unsigned int partID = 0;
+
+	/* Find id for a new partition */
+	if (strg->parent != NULL) {
+		part = strg->parent->parts;
+		if (part != NULL) {
+			do {
+				partID++;
+				part = part->next;
+			} while (part != strg->parent->parts);
+		}
+	}
+
+	/* Add mtdchar device */
+	if (strg->dev->mtd != NULL) {
+		oid->id &= ~MTD_MASK;
+		oid->id |= MTD_CHAR;
+
+		res = snprintf(path, sizeof(path), "/dev/mtd%u", strg->dev->id);
+		if (strg->parent != NULL) {
+			snprintf(path + res, sizeof(path), "p%u", partID);
+		}
+
+		res = create_dev(oid, path);
+		if (res < 0) {
+			fprintf(stderr, "zynq7000-flash: failed to create a device file, err: %d\n", res);
+			return res;
+		}
+	}
+
+	/* Add mtdblock device */
+	if (strg->dev->blk != NULL) {
+		oid->id &= ~MTD_MASK;
+		oid->id |= MTD_BLOCK;
+
+		res = snprintf(path, sizeof(path), "/dev/mtdblock%u", strg->dev->id);
+		if (strg->parent != NULL) {
+			snprintf(path + res, sizeof(path), "p%u", partID);
+		}
+
+		res = create_dev(oid, path);
+		if (res < 0) {
+			fprintf(stderr, "zynq7000-flash: failed to create a device file, err: %d\n", res);
+			return res;
+		}
+	}
+
+	return EOK;
+}
+
+
+static int flash_partAdd(const char *parentPath, off_t start, size_t size)
+{
+	int err;
+	oid_t poid, oid;
+	storage_t *strg, *parent;
+
+	err = flash_oidResolve(parentPath, &poid);
+	if (err < 0) {
+		printf("zynq7000-flash: cannot resolve %s\n", parentPath);
```

Comment:
In other places you use `fprintf(stderr, ...)`.
```suggestion
		fprintf(stderr, "zynq7000-flash: cannot resolve %s\n", parentPath);
```

Reply:
Yes, the message log should use`fprintf(stderr, ...)`, oversight.